### PR TITLE
[android] custom back button handler

### PR DESCRIPTION
**Pull request by skv-headless**
sometimes it is nessesary to handle back button
specifically for component, by changing its state.
 For ex. exit from edit mode.

### CODE REVIEW

**Comment by astreet** [comment]
Trying to coordinate the order that back press listeners are added would be very bug-prone... I think @ericvicenti is revisiting this soon, what are your thought Eric?

File: Libraries/Utilities/BackAndroid.android.js [hunk 1]
```diff
@@ -21,16 +21,19 @@ type BackPressEventName = $Enum<{
   backPress: string;
 }>;
 
-var _backPressSubscriptions = new Set();
+var _backPressSubscriptions = [];
```
Comment (mkonicek) [inline Libraries/Utilities/BackAndroid.android.js]: Why an array instead of a Set?

Comment (skv-headless) [inline Libraries/Utilities/BackAndroid.android.js]: I need reverse handlers execution (last registered - first executed). As I know Set doesn't guarantee order.

**Comment by skv-headless** [comment]
@astreet I agree with you, but for my case there are no more than 2 handlers at once (not so complex). Anyway I'm open to dialog. Suggest another API.

**Comment by mkonicek** [comment]
Assigning to @ericvicenti to review.

@skv-headless Can you rebase please? Sorry this got forgotten about, we are overwhelmed.

**Comment by skv-headless** [comment]
@ericvicenti I've rebased it

**Comment by ericvicenti** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**